internal/artifacts/module_arch.go

```diff
@@ -464,12 +464,18 @@ func archiveModuleToResultDir(buildResult string, requestedResultFileName string
 }
 
 // getIgnores - get files and/or subfolders to exclude from the package.
-func getIgnores(module *mta.Module) []string {
+func getIgnores(moduleLoc dir.ITargetModule, module *mta.Module) []string {
 	var ignoreList []string
 	// ignore defined in build params is declared
 	if module.BuildParams != nil && module.BuildParams[ignore] != nil {
 		ignoreList = convert(module.BuildParams[ignore].([]interface{}))
 	}
+	// we add temporary folder to the list of ignores, because
+	// for the case when source == target and module path == project path,
+	// temporary folder can become part of the packaged module content
+	tmpFolder := filepath.Dir(moduleLoc.GetTargetModuleDir(module.Name))
```

Comment:
There are 3 cases: project build, module build when not sending target and module build when sending target.
`GetTargetTmpDir` returns the expected temp dir for project build (the one with a sub folder for each module), and for module build when sending the target folder (the one where all the module build results will be). I think it doesn't cover the module build when you don't send the target (it will return the inner folder, the one with the module name). I think you will also have to add a parameter of whether the temp dir is the parent of the target folder.
To use `GetTargetTempDir` you will have to change the `IModule` interface to also implement `ITargetPath` (and maybe add the missing methods to the test class that implements it).

We should add in the documentation that the target folder will not be packaged in the build results, so it cannot be a folder with module files.